src/GafferSceneUI/VisualiserTool.cpp

```diff
@@ -994,6 +1014,34 @@ class VisualiserGadget : public Gadget
 								false /* throwIfInvalid */
 							);
 							break;
+						case FloatVectorDataTypeId :
+							vData = primitive->expandedVariableData<FloatVectorData>(
+								primVarName,
+								IECoreScene::PrimitiveVariable::Vertex,
+								false /* throwIfInvalid */
+							);
+							break;
+						case V2fVectorDataTypeId :
+							vData = primitive->expandedVariableData<V2fVectorData>(
+								primVarName,
+								IECoreScene::PrimitiveVariable::Vertex,
+								false /* throwIfInvalid */
+							);
+							break;
+						case V3fVectorDataTypeId :
+							vData = primitive->expandedVariableData<V3fVectorData>(
+								primVarName,
+								IECoreScene::PrimitiveVariable::Vertex,
+								false /* throwIfInvalid */
+							);
+							break;
+						case Color3fVectorDataTypeId :
+							vData = primitive->expandedVariableData<Color3fVectorData>(
+								primVarName,
+								IECoreScene::PrimitiveVariable::Vertex,
+								false /* throwIfInvalid */
+							);
+							break;
```

Comment:
This seems rather longwinded when you consider that `vData` is just a `ConstDataPtr`, so we've thrown away all the type information we got from the templated calls to `expandedVariableData()`. Can we just use one `expandedVariableData<Data>()` call after checking the types more concisely?

Reply:
Yes, that's much better now in https://github.com/GafferHQ/gaffer/pull/6212/commits/398f9a3a7023008e0b937c39a9deb379f75dbefd.